### PR TITLE
[CI] Add a workflow for quick perf comparison

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96166

Summary: ciflow/inductor-perf-test-nightly now contains full dashboard
run which takes a very long time. Ed proposed a simplification of the
perf run there, but it is still worth to have a set of fast perf test
which only includes one configuration (--training --amp).

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

### CODE REVIEW

**Comment by malfet** [comment]
@desertfire looks like it's a chicken and egg problem right now: bot only recognizes labels that exists on trunk. I'll try to submit a fix sometime next week, but in the meantime, just push the tag by hand

**Comment by weiwangmeta** [comment]
We now have small/medium/large inductor tests with this PR, great! 
To support this medium workflow, we may need to monitor the capacity. Since the nightly will trigger towards the end of the day, it may be beneficial to increase the (linux.gcp.a100.large) runner pool size from 4 to 8, as after around 6pm/7pm, this job would be queueing for a long time.

File: .github/workflows/upload-test-stats.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Upload test stats
 
 on:
   workflow_run:
-    workflows: [pull, trunk, periodic, inductor, inductor-A100-perf]
```
Comment (huydhn) [inline .github/workflows/upload-test-stats.yml:5]: You don't need `inductor-A100-perf-nightly` both in `upload-test-stats.yml` and in `upload-torch-dynamo-perf-stats`.  Once you have the commit in https://github.com/pytorch/pytorch/pull/96165, you will just need to update the latter `upload-torch-dynamo-perf-stats.yml`

**Comment by desertfire** [comment]
> We now have small/medium/large inductor tests with this PR, great! To support this medium workflow, we may need to monitor the capacity. Since the nightly will trigger towards the end of the day, it may be beneficial to increase the (linux.gcp.a100.large) runner pool size from 4 to 8, as after around 6pm/7pm, this job would be queueing for a long time.

For now, I have changed my PR to use the linux.gcp.a100 runner pool. Given that smoke tests are small and this perf comparison workflow is only triggered by manually adding the label, we should be ok.

**Review by huydhn — APPROVED** [review]

**Comment by weiwangmeta** [comment]
> 

Sure, I will monitor the queue and increase the max capacity if needed.

**Review by weiwangmeta — APPROVED** [review]
LGTM. How long do we expect the end to end time assuming 4 parallel runners? Is it still about 1 hour each?

**Comment by desertfire** [comment]
> LGTM. How long do we expect the end to end time assuming 4 parallel runners? Is it still about 1 hour each?

Yes, the run is within 1 hour.
